File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.cpp
using namespace std;

class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
        int n = chalk.size();
        long long int total_chalk = 0;
        for(int i = 0; i<n; i++){
            total_chalk += chalk[i];
        }
        k = k % total_chalk;
        
        
        int index = 0;
        while(1){
            if (k < chalk[index]){
                return index;
            }
            k -= chalk[index];
            index = (index + 1) % n;
        }
            
        
        return index;   
    }
};

File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        n = len(chalk)
        total_chalk = sum(chalk)
        k = k % total_chalk
        
        
        index = 0
        while True:
            if k < chalk[index]:
                return index
            k -= chalk[index]
            index = (index + 1) % n
        
        return index